### ISSUE TITLE
A leg not being closed down when B ends call

### ISSUE THREAD

**Issue by mase2hot** [signalwire/freeswitch#1293]
Hi All

I have an issue where the BYE is not being sent to the A leg once the B leg closes down the call. Now this only happens on calls where there were multiple re-invites caused by the B leg putting the far end on hold. When this doesn't take place there is only 1 re-invite from the B leg and the call closes down fine.

This is happening 99% of the time, I managed to grab a good and bad call. I can not see any obvious errors and FS is reporting that it will be sending a BYE to the A leg, but it does not. Please see attached debug from FS.

From what I can see on the call that fails to send the BYE, there are some state change extra things going on, no idea why?

Package version or git hash
 - Version 1.10.6

processor       : 23
vendor_id       : GenuineIntel
cpu family      : 6
model           : 63
model name      : Intel(R) Xeon(R) CPU E5-2620 v3 @ 2.40GHz
stepping        : 2
microcode       : 0x31
cpu MHz         : 1201.312
cache size      : 15360 KB
physical id     : 1
siblings        : 12
core id         : 5
cpu cores       : 6
apicid          : 27
initial apicid  : 27
fpu             : yes
fpu_exception   : yes
cpuid level     : 15
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm abm cpuid_fault epb invpcid_single pti tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm xsaveopt cqm_llc cqm_occup_llc dtherm ida arat pln pts
bugs            : cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs itlb_multihit
bogomips        : 4800.08
clflush size    : 64
cache_alignment : 64
address sizes   : 46 bits physical, 48 bits virtual
power management:
root@FS-4:~# uname -a
Linux FS-4 4.19.0-17-amd64 #1 SMP Debian 4.19.194-3 (2021-07-18) x86_64 GNU/Linux

Working = https://pastebin.com/yEu91AhZ

Not Working = https://pastebin.com/Hs67Xesa

**Comment by mase2hot** [comment]
I have discovered what the issue is. The BYE message on the failed calls is failing due to the character length of content of the reason header inside the text.

REASON: Q.850;cause=16;text="87664051-72f4-44f9-819e-403e177ed13b;This conversation has ended as only one participant was remaining in the conversation"

105 Characters inside quotes = works

106 characters inside quotes = fail

After header manipulation from test environment we figured this out. I could not find anything to suggest 105 is a standard limit so would seem Freeswitch has this for some reason. Can we get this resolved?

**Comment by hnimminh** [comment]
Are you using UDP? What is the size of you sip message? 
I'm thinking these issue would be MTU size.

**Comment by mase2hot** [comment]
Yes UDP, its not MTU as we tried with more characters without quotes and it worked fine. Its specifically when there are more than 105 characters inside the quotes in that text = value.

**Comment by mase2hot** [comment]
Freeswitch responds to the B leg with 200 OK from the BYE but when it trys to generate the BYE to the a leg it fails. I can only presume there is a functional limit set for value inside quotes being 105 characters maximum. We have dont many many test calls minupulating the Reason header. This how we know for sure its the character limit inside the quotes.
